docs/architecture/adr-008-event-subscription.md

```diff
@@ -0,0 +1,178 @@
+# ADR 008: RPC Client Event Subscription Mechanism
+
+## Changelog
+* 2020-07-23: Initial draft
+
+## Context
+
+The [Tendermint Light Client](../../light-client/) is one of many important
+applications that will make use of the [RPC client](../../rpc/) to query
+Tendermint full nodes for information relating to the blockchain.
+
+Tendermint servers (e.g. full nodes) provide an [event
+subscription][tm-event-subs] RPC endpoint to allow clients to receive
+notifications of specific events as they happen. We need to expose this
+subscription mechanism to the RPC client.
```

Comment:
Maybe here or in the title mention that this is about the websocket client particular?